awx_collection/plugins/modules/schedule.py

```diff
@@ -265,15 +265,29 @@ def main():
         search_fields['name'] = unified_job_template
         unified_job_template_id = module.get_one('unified_job_templates', **{'data': search_fields})['id']
         sched_search_fields['unified_job_template'] = unified_job_template_id
+
     # Attempt to look up an existing item based on the provided data
-    existing_item = module.get_one('schedules', name_or_id=name, **{'data': sched_search_fields})
+    existing_item = module.get_one('schedules', name_or_id=name, check_exists=(state == 'exists'), **{'data': sched_search_fields})
 
+    # set the default for enforced defaults
+    enforced_defaults = False
+    if state == 'absent':
+        # If the state was absent we can let the module delete it if needed, the module will handle exiting from this
+        module.delete_if_needed(existing_item)
+    elif state == 'enforced':
+        enforced_defaults = True
     association_fields = {}
 
+    # Fail early now if no unified job template provided now that enforced defaults set.
+    if unified_job_template is None and enforced_defaults:
+        module.fail_json(msg='When using enforced state unified_job_template is a required field')
+
     if credentials is not None:
         association_fields['credentials'] = []
         for item in credentials:
             association_fields['credentials'].append(module.resolve_name_to_id('credentials', item))
+    elif credentials is None and enforced_defaults:
+        association_fields['credentials'] = []
```

Comment:
@AlanCoding This line I tested removes existing credentials when submited to the API. when none are supplied and enforced is true.

Reply:
This would be the same as in a JT